Breeding-Insight/bi-api
# BI-1651 - Allow programs to choose/create breeding methods

# Description
**Story:** https://breedinginsight.atlassian.net/browse/BI-1651

Created a new set of tables and functionality to allow a program to defined their own breeding methods, and to specify which of the system-defined breeding methods they want to utilize in their program


# Dependencies
bi-web/feature/BI-1651 (https://github.com/Breeding-Insight/bi-web/pull/292)

# Testing
1. Verify that the flyway script runs successfully to create the new tables
2. See testing in https://github.com/Breeding-Insight/bi-web/pull/292

# Checklist:

- [ ] I have run TAF: https://github.com/Breeding-Insight/taf/actions/runs/3887368051
